### PR TITLE
Merge KICK:

### LINKED ISSUE #8: Command: KICK
Parameters: \<channel\> *( "," \<channel\> ) \<user\> *( "," \<user\> ) [\<comment\>]

Used to request the forced removal of a user from a channel.  It causes the \<user\> to PART from the \<channel\> by force.  For the message to be syntactically correct, there MUST be either one channel parameter and multiple user parameter, or as many channel parameters as there are user parameters.  If a "comment" is given, this will be sent instead of the default message, the nickname of the user issuing the KICK.

The server MUST NOT send KICK messages with multiple channels or users to clients.  This is necessarily to maintain backward compatibility with old client software.

Examples:

   KICK &Melbourne Matthew    _; Command to kick Matthew from &Melbourne_

   KICK #Finnish John :Speaking English   _; Command to kick John from #Finnish
                                   using "Speaking English" as the reason (comment)._

   :WiZ!jto@tolsun.oulu.fi KICK #Finnish John  _; KICK message on channel #Finnish from WiZ to remove John from channel_

ERROR replies:
- **ERR_NEEDMOREPARAMS** - 461
"\<command\> :Not enough parameters"
         -> Returned by the server by numerous commands to indicate to the client that it didn't supply enough parameters.

- **ERR_NOSUCHCHANNEL** - 403
"\<channel name\> :No such channel"
         -> Used to indicate the given channel name is invalid.

- **ERR_NOTONCHANNEL** - 442
"\<channel\> :You're not on that channel"
         -> Returned by the server whenever a client tries to perform a channel affecting command for which the client isn't a member.

- **ERR_BADCHANMASK** - 476
"\<channel\> :Bad Channel Mask"
         -> Any command requiring operator privileges to operate MUST return this error to indicate the attempt was unsuccessful.

- **ERR_CHANOPRIVSNEEDED** - 482
"\<channel\> :You're not channel operator"
         -> Any command requiring 'chanop' privileges (such as MODE messages) MUST return this error if the client making the attempt is not a chanop on the specified channel.

- **ERR_USERNOTINCHANNEL** - 441
"\<nick\> \<channel\> :They aren't on that channel"
         -> Returned by the server to indicate that the target user of the command is not on the given channel.

### PR DESCRIPTION
**Pull request by Elena-Lou**
Confirm() is added to KICK class.

_foundChannel is added as a class attribute to be passed on to confirm() and then used to broadcast the msg to all the clients connected to that particular channel

Close #8 

Co-authored-by: aweaver [aweaver@student.42.fr](mailto:aweaver@student.42.fr)
Co-authored-by: elouisia [elouisiade@live.fr](mailto:elouisiade@live.fr)

### CODE REVIEW

File: Kick.cpp [hunk 1]
```diff
@@ -11,8 +11,10 @@ Kick::Kick( Kick const & src ) : ACommand(src)
 
 Kick::Kick(Server &server, Client &user, std::string rawInput) : ACommand(server, user, rawInput)
 {
+
 	std::cout << "KICK overloaded constructor" << std::endl;
 	std::cout << "client socketFD : " << this->_author->getSocketFD() << std::endl;
```
Comment (AdrianWeaver) [inline Kick.cpp:16]: Please encapsulate in
#if SHOW_CONSTRUCTOR
#endif 

(same for destructor)

**Review by AdrianWeaver — CHANGES_REQUESTED** [review]
Great work mate 🤓 !
I cannot wait to have join to test if all this works as expected !
Will still validate this like that once you add the compiler instructions in constructor/destructor messages, so we can move on and reopen the Kick issue if needed later.

File: Kick.cpp (hunk 1 above)
Comment (Elena-Lou) [reply Kick.cpp:19]: Thank you for your input

**Review by AdrianWeaver — APPROVED** [review]
Noice ! 👍

**Pull request merged**